### PR TITLE
WAIT RzType refactoring: Refactor rz_analysis_dwarf_integrate_functions

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

~Rizin report warning messages increasingly when repeated run aaa.~
~This is due to the fact that analysis information aren't reset between~
~2 aaa analysis so the blocks analysis isn't deterministic.~

~The simple 'hack' is to force the fcns analysis data to be reset.~

~I add a simple test to verify that the second analysis isn't overriding~
~the first one.~

~I believe that if this ´solution´ is accepted, we should update the aaa help.~

~And the test suite will fail, because I didn´t add the binary to https://github.com/rizinorg/rizin-testbins. I would like to have some advice on this first.~

**Test plan**

see  #329 and the new test

**Closing issues**

 #329

### CODE REVIEW

**Comment by ghost** [comment]
I believe that the problem come from the function rz_analysis_dwarf_integrate_functions

File: librz/analysis/dwarf_process.c [hunk 1]
```diff
@@ -1571,75 +1688,13 @@ RZ_API void rz_analysis_dwarf_integrate_functions(RzAnalysis *analysis, RzFlag *
...
-				rz_analysis_function_set_var(fcn, offset - fcn->bp_off, *kind, type, 4, false, var_name);
-			}
-			free(var_key);
-			free(var_data);
-		loop_end:
-			sdb_aforeach_next(var_name);
-		}
-		free(var_names_key);
-		free(vars);
+		RzAnalysisFunction *fcn = get_fcn_from_sname(analysis, dwarf_sdb, func_sname);
```
Comment (ret2libc) [inline librz/analysis/dwarf_process.c:1744]: I believe the check `if (fcn)` should be here and not in the `update_fcn_*` functions.

**Review by ret2libc — COMMENTED** [review]
When you make the PR ready for review, please make sure to update the description to mention where to look exactly for the fix. I see you refactored the code (much better now, thanks!) but it's not clear if/where is the fix.

**Comment by ghost** [comment]
> When you make the PR ready for review, please make sure to update the description to mention where to look exactly for the fix. I see you refactored the code (much better now, thanks!) but it's not clear if/where is the fix.

Okay, right now I am searching.

**Comment by ghost** [comment]
I don't have time to work on this bug. However the refactoring could be merged.

**Review by ret2libc — APPROVED** [review]

**Comment by ghost** [comment]
Wait RzType refactoring before merging

**Review by wargio — CHANGES_REQUESTED** [review]
requires another PR to be merged first

**Comment by ghost** [comment]
Removed because it make no sense with the new rz_type refactoring.

**Pull request closed without merging**